CollAction/Services/ProjectService.cs

```diff
@@ -64,26 +72,40 @@ public ProjectService(ApplicationDbContext context, IStringLocalizer<IProjectSer
         {
             return await _context.Projects
-               .Where(filter)
-               .OrderBy(p => p.DisplayPriority)
-               .Select(p =>
-                new DisplayTileProjectViewModel(_localizer, p.Start, p.End, p.Status, p.IsActive, p.IsComingSoon, p.IsClosed)
-                {
-                    ProjectId = p.Id,
-                    ProjectName = p.Name,
-                    ProjectProposal = p.Proposal,
-                    CategoryName = p.Category.Name,
-                    CategoryColorHex = p.Category.ColorHex,
-                    LocationName = p.Location.Name,
-                    BannerImagePath = p.BannerImage != null ? p.BannerImage.Filepath : $"/images/default_banners/{p.Category.Name}.jpg",
-                    BannerImageDescription = p.BannerImage.Description,
-                    Target = p.Target,
-                    Participants = p.Participants.Sum(participant => participant.User.RepresentsNumberParticipants) + p.AnonymousUserParticipants
-                })
-               .ToListAsync();
+                .Where(filter)
+                .Include(p => p.Category)
+                .Include(p => p.Location)
+                .Include(p => p.BannerImage)
+                .Include(p => p.DescriptiveImage)
+                .Include(p => p.DescriptionVideoLink)
+                .Include(p => p.Owner)
+                .Include(p => p.Tags).ThenInclude(t => t.Tag)
+                .GroupJoin(_context.ProjectParticipants,
+                    project => project.Id,
+                    participants => participants.ProjectId,
+                    (project, participantsGroup) => new FindProjectsViewModel(_stringLocalizer)
+                    {
+                        ProjectId = project.Id,
+                        ProjectName = project.Name,
+                        ProjectProposal = project.Proposal,
+                        CategoryName = project.Category.Name,
+                        CategoryColorHex = project.Category.ColorHex,
+                        LocationName = project.Location.Name,
+                        BannerImagePath = project.BannerImage != null ? project.BannerImage.Filepath : $"/images/default_banners/{project.Category.Name}.jpg",
+                        BannerImageDescription = project.BannerImage.Description,
+                        Target = project.Target,
+                        Participants = project.Participants.Sum(participant => participant.User.RepresentsNumberParticipants) + project.AnonymousUserParticipants,
+                        Remaining = project.RemainingTime,
+                        DescriptiveImagePath = project.DescriptiveImage.Filepath,
+                        DescriptiveImageDescription = project.DescriptiveImage.Description,
+                        Status = project.Status,
+                        Start = project.Start,
+                        End = project.End
+                    })
+                .ToListAsync();
```

Comment:
This probably generates a very complex query, in a method that's called a lot. You might want to profile it, see if the database can take it.

In my experience, a limited set if simple queries can perform a lot better then a very complex one, even if it does require more roundtrips.

Reply:
True, we should do some optimization.

This however isn't a new query, it's the exact same query that was already there, things just moved around a little.

Maybe we should create a seperate ticket for that, I didn't change anyhere except the naming of the viewmodels.